idaes/core/util/model_diagnostics.py

```diff
@@ -17,29 +17,851 @@
+        stream,
+        lines_list,
+        title=None,
+        else_line=None,
+        end_line=None,
+        header="-",
+        footer=None,
+    ):
+        """
+        Writes output in standard format for report and display methods.
+
+        Args:
+            stream: stream to write to
+            lines_list: list containing lines to be written in body of report
+            title: title to be put at top of report
+            else_line: line to be written if lines_list is empty
+            end_line: line to be written at end of report
+            header: character to use to write header separation line
+            footer: character to use to write footer separation line
+
+        Returns:
+            None
+
+        """
+        stream.write(f"{header * MAX_STR_LENGTH}\n")
+        if title is not None:
+            stream.write(f"{title}\n\n")
+        if len(lines_list) > 0:
+            for i in lines_list:
+                stream.write(f"{TAB}{i}\n")
+        elif else_line is not None:
+            stream.write(f"{TAB}{else_line}\n")
+        stream.write("\n")
+        if end_line is not None:
+            stream.write(f"{end_line}\n")
+        if footer is not None:
+            stream.write(f"{footer * MAX_STR_LENGTH}\n")
+
+    def report_structural_issues(self, stream=stdout):
+        """
+        Generates a summary report of any structural issues identified in the model provided
+        and suggest next steps for debugging model.
+
+        This should be the first method called when debugging a model and after any change
+        is made to the model. These checks can be run before trying to initialize and solve
+        the model.
+
+        Args:
+            stream: I/O object to write report to (default = stdout)
+
+        Returns:
+            None
+
+        """
+        # Potential evaluation errors
+        # High Index
+
+        vars_in_constraints = variables_in_activated_constraints_set(self.model)
+        fixed_vars_in_constraints = ComponentSet()
+        free_vars_in_constraints = ComponentSet()
+        free_vars_lb = ComponentSet()
+        free_vars_ub = ComponentSet()
+        free_vars_lbub = ComponentSet()
+        ext_fixed_vars_in_constraints = ComponentSet()
+        ext_free_vars_in_constraints = ComponentSet()
+        for v in vars_in_constraints:
+            if v.fixed:
+                fixed_vars_in_constraints.add(v)
+                if not _var_in_block(v, self.model):
+                    ext_fixed_vars_in_constraints.add(v)
+            else:
+                free_vars_in_constraints.add(v)
+                if not _var_in_block(v, self.model):
+                    ext_free_vars_in_constraints.add(v)
+                if v.lb is not None:
+                    if v.ub is not None:
+                        free_vars_lbub.add(v)
+                    else:
+                        free_vars_lb.add(v)
+                elif v.ub is not None:
+                    free_vars_ub.add(v)
```

Comment:
This seems like a lot of logic for a method that is presumably just doing some reporting. I guess it is not used elsewhere?